From 680243c3d6f845d4dc0936c347cf1cdf2301c083 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 23 Dec 2025 15:10:48 +0800
Subject: [PATCH] Fix build for 32bit platforms

Fix the following build errors:

In file included from pesign.h:18,
                 from cms_common.c:21:
cms_common.c: In function 'cms_set_pw_data':
util.h:275:23: error: format '%ld' expects argument of type 'long int', but argument 2 has type '__time64_t' {aka 'long long int'} [-Werror=format=]
  275 |                 warnx("%ld.%lu %s:%s():%d: " fmt,               \
      |                       ^~~~~~~~~~~~~~~~~~~~~~
util.h:275:23: error: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type '__suseconds64_t' {aka 'long long int'} [-Werror=format=]
  275 |                 warnx("%ld.%lu %s:%s():%d: " fmt,               \
      |                       ^~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Backport [https://salsa.debian.org/efi-team/pesign/-/raw/debian/debian/patches/t64-fix.patch]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 src/efikeygen.c | 2 +-
 src/util.h      | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/efikeygen.c b/src/efikeygen.c
index 74e6011..32c5d4b 100644
--- a/src/efikeygen.c
+++ b/src/efikeygen.c
@@ -1163,7 +1163,7 @@ int main(int argc, char *argv[])
 	} else {
 		// Mon Jan 19 03:14:07 GMT 2037, aka 0x7fffffff minus 1 year.
 		time_t time = 0x7ffffffful - 60ul * 60 * 24 * 365;
-		dbgprintf("not_valid_after:%lu", time);
+		dbgprintf("not_valid_after:%llu", (unsigned long long)time);
 		not_after = (PRTime)time * PR_USEC_PER_SEC;
 	}
 	dbgprintf("not_after:%"PRId64, not_after);
diff --git a/src/util.h b/src/util.h
index 6616011..e22789f 100644
--- a/src/util.h
+++ b/src/util.h
@@ -272,8 +272,8 @@ extern long verbosity(void);
 #define dbgprintf_(tv, file, func, line, fmt, args...) ({	\
 		struct timeval tv;				\
 		gettimeofday(&tv, NULL);			\
-		warnx("%ld.%lu %s:%s():%d: " fmt,		\
-		      tv.tv_sec, tv.tv_usec,			\
+		warnx("%lld.%llu %s:%s():%d: " fmt,		\
+		      (long long)tv.tv_sec, (unsigned long long)tv.tv_usec,			\
 		      file, func, line, ##args);		\
 	})
 #if defined(PESIGN_DEBUG)
-- 
2.34.1

